services/admin/src/admin/authentication.py

```diff
@@ -46,11 +49,27 @@ async def auth_check(
             if organization is None or (
                 organization in {org["name"] for org in json["orgs"]}
                 and json["auth"]["type"] == "access_token"
-                and "fineGrained" in json["auth"]["accessToken"]
-                and any(
-                    set(permission for permission in scope["permissions"]) >= set(require_fine_grained_permissions)
-                    for scope in json["auth"]["accessToken"]["fineGrained"]["scoped"]
-                    if scope["entity"]["name"] == organization and scope["entity"]["type"] == "org"
+                and (
+                    (
+                        "fineGrained" in json["auth"]["accessToken"]
+                        and any(
+                            set(permission for permission in scope["permissions"])
+                            >= set(require_fine_grained_permissions)
+                            for scope in json["auth"]["accessToken"]["fineGrained"]["scoped"]
+                            if scope["entity"]["name"] == organization and scope["entity"]["type"] == "org"
+                        )
+                    )
+                    or (
+                        "role" in json["auth"]["accessToken"]
+                        and any(
+                            json["auth"]["accessToken"]["role"] == require_token_and_org_permission[0]
+                            and any(
+                                org["name"] == organization and org["roleInOrg"] == require_token_and_org_permission[1]
+                                for org in json["orgs"]
+                            )
+                            for require_token_and_org_permission in require_token_and_org_permissions
+                        )
+                    )
```

Comment:
this simply allows for write tokens, which is stronger than fine-grained. write tokens are used in the ci so this fixes ci debugging